src/Microsoft.Health.Fhir.CodeGen/CompareTool/PackageComparer.cs

```diff
@@ -1402,6 +1413,244 @@ private void WriteValueSetMaps(string outputDir, IEnumerable<ValueSetComparison>
+            sm.Name = sm.Id;
+            sm.Url = $"http://hl7.org/fhir/uv/xver/StructureMap/{sm.Id}";
+            sm.Title = $"{c.Source.Name} Transforms: {srcVersion.ToRLiteral()} to {tgtVersion.ToRLiteral()}";
+            sm.Description = c.Message;
+
+            sm.Import = [$"http://hl7.org/fhir/uv/xver/StructureMap/*{srcVersion.ToRLiteral().TrimStart('R')}to{tgtVersion.ToRLiteral().TrimStart('R')}"];
+
+            // Create the uses structures
+            sm.Structure.Add(new StructureMap.StructureComponent()
+            {
+                Mode = StructureMap.StructureMapModelMode.Source,
+                Alias = srcResourceType + srcVersion.ToRLiteral(),
+                Url = srcScope.Uri?.Replace("http://hl7.org/fhir/StructureDefinition/", $"http://hl7.org/fhir/{srcVersion.ToShortVersion()}/StructureDefinition/"),
+            });
+            sm.Structure.Add(new StructureMap.StructureComponent()
+            {
+                Mode = StructureMap.StructureMapModelMode.Target,
+                Alias = tgtResourceType + tgtVersion.ToRLiteral(),
+                Url = tgtScope.Uri?.Replace("http://hl7.org/fhir/StructureDefinition/", $"http://hl7.org/fhir/{tgtVersion.ToShortVersion()}/StructureDefinition/"),
+            });
+
+            // Create the groups
+            var propComparisons = c.ElementComparisons.ToList();
+            // var propComparisons = c.ElementComparisons.OrderBy(kvp => kvp.Key).ToList();
+            foreach ((string path, ElementComparison elementComparison) in propComparisons)
+            {
+                if (elementComparison.Source.Types.Any(t => t.Value.Name == "BackboneElement") || !elementComparison.Source.Path.Contains("."))
+                {
+                    var group = new StructureMap.GroupComponent();
+                    group.Name = String.Join("", elementComparison.Source.Path.Split('.').Select(p => p.Substring(0, 1).ToUpper() + p.Substring(1))); // need to convert the path into a better name than this so you get AccountGuarantor
+                    group.Input.Add(new StructureMap.InputComponent()
+                    {
+                        Mode = StructureMap.StructureMapInputMode.Source,
+                        Name = "src",
+                        // Type = elementComparison.Source.Types
+                    });
+                    group.Input.Add(new StructureMap.InputComponent()
+                    {
+                        Mode = StructureMap.StructureMapInputMode.Target,
+                        Name = "tgt",
+                        // Type = elementComparison.Source.Types
+                    });
+                    if (!elementComparison.Source.Path.Contains("."))
+                    {
+                        group.Input[0].Type = srcResourceType + srcVersion.ToRLiteral();
+                        group.Input[1].Type = tgtResourceType + tgtVersion.ToRLiteral();
+                        group.Extends = "Resource";
```

Comment:
I couldn't spot the routine to retrieve the Base type for the structure being processed in the resource.
(searching for DomainResource or Resource, or the other datatypes where they are for complex types)

Reply:
In the `StructureComparison` object (`c`, I believe at this point in the code), there are references to the `Source` and `Target` `StructureDefinition` resources subject to comparison.

In each of the structures, there is the function to get the base type (`cgBaseTypeName`).

In `DefinitionCollection`, there are a few functions you could use - I would probably lean towards `TryGetStructure`.

If we want a collapsed convenience function to do both, I would be fine with that (e.g., `cgBaseTypeStructure`).